src/API/FeedState.php

```diff
@@ -261,7 +262,29 @@ private function add_feed_registration_state( $result ) {
 				throw new \Exception( esc_html__( 'Could not get merchant info.', 'pinterest-for-woocommerce' ) );
 			}
 
-			if ( 'ACTIVE' !== $merchant['data']->product_pin_feed_profile->feed_status ) {
+			$feed = Base::get_merchant_feeds( $merchant_id );
```

Comment:
Instead of getting all the feeds here and having the logic that gets the single feed in here, we should have a `get_merchant_feed()` method that returns the single feed for the given feed_id